Content.Server/Spawners/EntitySystems/GenericSpawnerSystem.cs

```diff
@@ -73,31 +73,32 @@ private void Spawn(EntityUid uid, GenericSpawnerComponent component)
         if (component.EntityTables.Count == 0)
             return;
 
-        foreach (var pair in component.EntityTables)
-        {
-            if (!_proto.TryIndex(pair.Key, out var entTable))
-            {
-                Log.Warning($"Referenced entity table prototype does not exist! Entity: {ToPrettyString(uid)}");
-                continue;
-            }
 
-            if (pair.Value != 1.0f && !_robustRandom.Prob(pair.Value))
-                continue;
+        if (component.Rolls is < 1 or > 100)
+        {
+            Log.Warning($"Invalid amount of rolls on entity table, value should be between 1 and 100. Entity: {ToPrettyString(uid)}");
+            return;
+        }
 
-            if (entTable.Weights.Count == 0)
+        foreach (var _ in Enumerable.Repeat(1, component.Rolls))
+        {
+            foreach (var pair in component.EntityTables)
             {
-                Log.Warning($"Entity table in GenericSpawnerComponent is empty! Entity: {ToPrettyString(uid)}");
-                continue;
-            }
+                if (!_proto.TryIndex(pair.Key, out var entTable))
```

Comment:
move the TryIndex for the `key` below the probability check, we don't need to check if the proto is valid every single time if it's just going to fail the probability check on the `value` anyways